laa_crime_application_store_app/routers/application.py

```diff
@@ -83,3 +85,51 @@ async def post_application(request: ApplicationNew, db: Session = Depends(get_db
         nested.rollback()
         return Response(status_code=409)
     return Response(status_code=201)
+
+
+@router.put("/application/{app_id}", status_code=201, responses=responses)
+async def put_application(
+    app_id: UUID, request: ApplicationUpdate, db: Session = Depends(get_db)
+):
+    application = db.query(Application).filter(Application.id == app_id).first()
+    if application is None:
+        logger.info("APPLICATION_NOT_FOUND", application_id=app_id)
+        return Response(status_code=400)
+
+    application_version = (
+        db.query(ApplicationVersion)
+        .filter(
+            ApplicationVersion.application_id == app_id,
+            ApplicationVersion.version == application.current_version,
+        )
+        .first()
+    )
+
+    # we ignore the issue if no application version is found to
+    # avoid the system getting into a state where new versions
+    # can't be added.
+    if (
+        application_version
+        and application_version.application == request.application
+        and application.application_state == request.application_state
+    ):
+        return Response(status_code=200)
```

Comment:
This should probably be a 204 - No Content

Reply:
updated